src/router/index.js

```diff
@@ -1,24 +1,21 @@
-import { parse, stringify } from "query-string";
+export function extractStoreFromLocation(query) {
+  try {
+    const matches = decodeURIComponent(query).match(/storeData=({.*})/, "");
```

Comment:
I think the inner components might need to be decoded in case they are encoded in addition to the global string. I usually do something like this:

```js
const matches = decodeURIComponent(query)
  .replace(/^\?/, '')
  .split('&')
  .map(pair => pair.split('='))
  .map(pair => pair.map(decodeURIComponent))
  .filter(([name, value]) => name == 'storeData');

if (matches.length > 0) {
  try {
    return JSON.parse(matches[0][1]);
  } catch(e) { return data; }
}

return {};

```

Reply:
Feel free to collapse that down for simplicity, just illustrating the point.

Reply:
I'm not sure I see the point here? The code decodes the global query and extracts `storeData` from the query and ignores everything else. Why would we need to decode the other parameters if we're not using them?